sql/dd/impl/bootstrap/bootstrapper.cc

```diff
@@ -736,8 +757,31 @@ bool DDSE_dict_init(THD *thd, dict_init_mode_t dict_init_mode, uint version) {
   List<const Plugin_tablespace> ddse_tablespaces;
   if (ddse->ddse_dict_init == nullptr ||
       ddse->ddse_dict_init(dict_init_mode, version, &ddse_tables,
-                           &ddse_tablespaces))
+                           &ddse_tablespaces)) {
+    LogErr(ERROR_LEVEL, ER_LOG_PRINTF_MSG,
+           "Calling DDSE handlerton ddse_dict_init failed");
     return true;
+  }
+
+  /*
+    Always call innodb handler API unless disabled
+    innobase_dict_init will initialize these innodb specific tables
+  */
+  if (ddse->db_type != DB_TYPE_INNODB) {
+    handlerton *innodb_ddse = ha_resolve_by_legacy_type(thd, DB_TYPE_INNODB);
+    bool innodb_disabled = ha_is_storage_engine_disabled(innodb_ddse);
+    if (!innodb_disabled) {
+      assert(!ddse_tablespaces.size());
+      if (innodb_ddse->ddse_dict_init == nullptr ||
+          innodb_ddse->ddse_dict_init(dict_init_mode, version, &ddse_tables,
+                                      &ddse_tablespaces)) {
+        LogErr(
+            ERROR_LEVEL, ER_LOG_PRINTF_MSG,
+            "Calling non-Default DDSE InnoDB handlerton ddse_dict_init failed");
```

Comment:
I am not sure about the wording here: the DDSE is singleton, and it is not InnoDB here, and this is not about upgrade scenario. So InnoDB here is not "a DDSE, but non-default one", IMHO I'd call it just a SE
```suggestion
            "Calling non-default DDSE InnoDB handlerton ddse_dict_init failed");
```

Reply:
just copy original words from your PR